internal/settings/ovh.go

```diff
@@ -142,3 +169,77 @@ func (o *ovh) Update(ctx context.Context, client *http.Client, ip net.IP) (newIP
 		return nil, fmt.Errorf("%w: %s", ErrUnknownResponse, s)
 	}
 }
+
+func (o *ovh) updateWithZoneDNS(client *ovhClient.Client, ip net.IP) (newIP net.IP, err error) {
+	recordType := A
+	var ipStr string
+	if ip.To4() == nil { // IPv6
+		recordType = AAAA
+		ipStr = ip.To16().String()
+	} else {
+		ipStr = ip.To4().String()
+	}
+	// subDomain filter of the ovh api expect an empty string to get @ record
+	subDomain := o.host
+	if subDomain == "@" {
+		subDomain = ""
+	}
+	// get existing records
+	var recordIDs []uint64
+	if err := client.Get(fmt.Sprintf("/domain/zone/%s/record?fieldType=%s&subDomain=%s", o.domain,
+		recordType, subDomain), &recordIDs); err != nil {
+		return ip, err
+	}
+	if len(recordIDs) == 0 {
+		// create a new record
+		postRecordsParams := struct {
+			FieldType string `json:"fieldType"`
+			SubDomain string `json:"subDomain"`
+			Target    string `json:"target"`
+		}{
+			FieldType: recordType,
+			SubDomain: subDomain,
+			Target:    ipStr,
+		}
+		if err := client.Post(fmt.Sprintf("/domain/zone/%s/record", o.domain), &postRecordsParams, nil); err != nil {
+			return ip, err
+		}
+	} else {
+		// update existing record
+		putRecordsParams := struct {
+			Target string `json:"target"`
+		}{
+			Target: ipStr,
+		}
+		for _, recordID := range recordIDs {
+			if err := client.Put(fmt.Sprintf("/domain/zone/%s/record/%d", o.domain, recordID),
+				&putRecordsParams, nil); err != nil {
+				return ip, err
+			}
+		}
+	}
+
+	if err := client.Post(fmt.Sprintf("/domain/zone/%s/refresh", o.domain), nil, nil); err != nil {
+		return ip, err
+	}
+
+	return ip, nil
+}
+
+func (o *ovh) Update(ctx context.Context, client *http.Client, ip net.IP) (newIP net.IP, err error) {
+	if o.zoneDNS {
+		if len(o.apiEndpoint) == 0 {
+			o.apiEndpoint = "ovh-eu"
+		}
+		ovhClientInstance, _ := ovhClient.NewClient(
+			o.apiEndpoint,
+			o.appKey,
+			o.appSecret,
+			o.consumerKey,
+		)
+		newIP, err = o.updateWithZoneDNS(ovhClientInstance, ip)
+	} else {
+		newIP, err = o.updateWithDynHost(ctx, client, ip)
+	}
+	return newIP, err
```

Comment:
Better to return early to have less indentation

```suggestion
	if !o.zoneDNS {
		return  o.updateWithDynHost(ctx, client, ip)
	}
	if len(o.apiEndpoint) == 0 {	
		o.apiEndpoint = "ovh-eu"
	}
	ovhClientInstance, _ := ovhClient.NewClient(
		o.apiEndpoint,
		o.appKey,
		o.appSecret,
		o.consumerKey,
	)
	return o.updateWithZoneDNS(ovhClientInstance, ip)
```